test/expression_test.ts

```diff
@@ -92,4 +92,90 @@ describe('expressions', () => {
     expectTranslate('foo.bar;').to.equal('foo.bar;');
     expectTranslate('foo[bar];').to.equal('foo[bar];');
   });
+  it('should cast numeric values to int when bitwise operators are used for a LHS exp RHS numeric expression',
+     () => {
+       function makeExpr(operator: string) {
+         return 'var x: number = 1; var y: number = 2; var z: number = x ' + operator + ' y;';
+       }
+
+       expectTranslate(makeExpr('|'))
+           .to.equal("num x = 1;\nnum y = 2;\nnum z = (x as int) | (y as int);");
```

Comment:
consider just having the `num z = (x as int) | (y as int)` part, but dropping the declarations before, I think they are not needed.